comment/utils.py

```diff
@@ -88,6 +101,11 @@ def get_comment_context_data(request, model_object=None, comments_per_page=0):
 
     allowed_flags = getattr(settings, 'COMMENT_FLAGS_ALLOWED', 0)
     oauth = request.POST.get('oauth') or request.GET.get('oauth')
+    if oauth == 'true':
+        oauth = True
+    else:
+        oauth = False
+    is_anonymous_allowed = getattr(settings, 'COMMENT_ALLOW_ANONYMOUS', False)
```

Comment:
Since `oauth` was coming from an HTTP request, it was a string. It would always be true in the template.
I think this was unintended.